### PR TITLE
Move user related funcs from common/dao

### PR DESCRIPTION
**Pull request by reasonerjt**
This commit moves more user related funcs, such as ChangePassword,
Login, ChangeUserProfile from common/dao to rely on /pkg/user and
pkg/oidc.
It also removes the code for resetting user's password as it's disabled.

Signed-off-by: Daniel Jiang <jiangd@vmware.com>

### CODE REVIEW

File: src/pkg/user/manager.go [hunk 1]
```diff
@@ -75,20 +76,33 @@ func (m *manager) Delete(ctx context.Context, id int) error {
...
+		return nil, err
 	}
-	return utils.Encrypt(password, u.Salt, u.PasswordVersion) == u.Password, nil
+	l = append(l, u)
+	for _, entry := range l {
+		if utils.Encrypt(password, entry.Salt, entry.PasswordVersion) == entry.Password {
+			return entry, nil
+		}
+	}
+	return nil, nil
```
Comment (heww) [inline src/pkg/user/manager.go:94]: There is a special char `@` in the email, it is invalid in the username.

**Review by wy65701436 — APPROVED** [review]
lgtm

File: src/pkg/user/manager.go (hunk 1 above)
Comment (reasonerjt) [reply src/pkg/user/manager.go:94]: But this is for querying?
I don't recall how the decision was made, but this is for compatibility with `LoginByDb` in `common/dao` pkg:
https://github.com/goharbor/harbor/blob/release-2.2.0/src/common/dao/user.go#L81

Comment (heww) [reply src/pkg/user/manager.go:94]: There are two db queries in this method `MatchLocalPassword `, only one db query in `LoginByDb`.

And when `usernameOrEmail` is a email, `u, err := m.GetByName(ctx, usernameOrEmail)` will must get a not found error.

**Review by heww — APPROVED** [review]

File: src/pkg/user/manager.go (hunk 1 above)
Comment (reasonerjt) [reply src/pkg/user/manager.go:94]: I have added a `FilterBy...` func to user model.

**Pull request merged**